### ISSUE TITLE
compile error: 'bpy::numeric' has not been declared

### ISSUE THREAD

**Issue by olimoser** [FORTH-ModelBasedTracker/PyOpenPose#23]
I am trying to build PyOpenPose on Ubuntu 16.04

```
~/work/PyOpenPose/build$ cmake ..
Install prefix is set to: /usr/local
Generating with build type: Release
-- Could NOT find Doxygen (missing:  DOXYGEN_EXECUTABLE) 

-- Boost version: 1.65.1
-- Found the following Boost libraries:
--   system
--   python-py35
-- 
-- The following REQUIRED packages have been found:

 * Threads
 * CUDA
 * OpenCV (required version >= 3)
 * Boost
 * PythonLibs (required version == 3)

-- The following OPTIONAL packages have not been found:

 * Doxygen

-- Configuring done
-- Generating done
-- Build files have been written to: /home/user/work/PyOpenPose/build
```

make fails with:

```
~/work/PyOpenPose/build$ make 
[ 25%] Building CXX object PyOpenPoseLib/CMakeFiles/PyOpenPose.dir/opencv_expose.cpp.o
/home/user/work/PyOpenPose/PyOpenPoseLib/opencv_expose.cpp: In function 'int initialize_array()':
/home/user/work/PyOpenPose/PyOpenPoseLib/opencv_expose.cpp:32:14: error: 'bpy::numeric' has not been declared
         bpy::numeric::array::set_module_and_type("numpy", "ndarray");
              ^
PyOpenPoseLib/CMakeFiles/PyOpenPose.dir/build.make:110: recipe for target 'PyOpenPoseLib/CMakeFiles/PyOpenPose.dir/opencv_expose.cpp.o' failed
make[2]: *** [PyOpenPoseLib/CMakeFiles/PyOpenPose.dir/opencv_expose.cpp.o] Error 1
CMakeFiles/Makefile2:117: recipe for target 'PyOpenPoseLib/CMakeFiles/PyOpenPose.dir/all' failed
make[1]: *** [PyOpenPoseLib/CMakeFiles/PyOpenPose.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
``` 

any ideas?

**Comment by padeler** [comment]
This looks like a boost-python version issue, can you try it with boost 1.58 (the default boost for ubuntu 16.04)?

**Comment by olimoser** [comment]
boost 1.58 is (also) installed
How can I force cmake/make to use this version?

**Comment by padeler** [comment]
Add the boost version in the find_package line in CMakeLists.txt, i.e. for python3 use:
```cmake
find_package(Boost 1.58 EXACT COMPONENTS system python-py35 REQUIRED)  
```

**Comment by olimoser** [comment]
I still get the same error.
I installed boost 1.58 from source into a seperate folder. 

my cmake command is:
`cmake -DBOOST_ROOT=/home/user/work/boost_1.58/ -DBOOST_LIBRARYDIR=/home/user/work/boost_1.58/lib/ ..`

```
cmake ..
Install prefix is set to: /usr/local
Generating with build type: Release
-- Could NOT find Doxygen (missing:  DOXYGEN_EXECUTABLE) 
-- Boost version: 1.58.0
-- Found the following Boost libraries:
--   system
--   python-py35
-- 
-- The following REQUIRED packages have been found:

 * Threads
 * CUDA
 * OpenCV (required version >= 3)
 * Boost (required version == 1.58)
 * PythonLibs (required version == 3.5.2)

-- The following OPTIONAL packages have not been found:

 * Doxygen

-- Configuring done
-- Generating done
-- Build files have been written to: /home/user/work/PyOpenPose/build
(cv):~/work/PyOpenPose/build$ make
Scanning dependencies of target PyOpenPose
[ 25%] Building CXX object PyOpenPoseLib/CMakeFiles/PyOpenPose.dir/OpenPoseWrapper.cpp.o
[ 50%] Building CXX object PyOpenPoseLib/CMakeFiles/PyOpenPose.dir/PyOpenPose.cpp.o
[ 75%] Building CXX object PyOpenPoseLib/CMakeFiles/PyOpenPose.dir/opencv_expose.cpp.o
/home/user/work/PyOpenPose/PyOpenPoseLib/opencv_expose.cpp: In function 'int initialize_array()':
/home/user/work/PyOpenPose/PyOpenPoseLib/opencv_expose.cpp:32:14: error: 'bpy::numeric' has not been declared
         bpy::numeric::array::set_module_and_type("numpy", "ndarray");
              ^
PyOpenPoseLib/CMakeFiles/PyOpenPose.dir/build.make:110: recipe for target 'PyOpenPoseLib/CMakeFiles/PyOpenPose.dir/opencv_expose.cpp.o' failed
make[2]: *** [PyOpenPoseLib/CMakeFiles/PyOpenPose.dir/opencv_expose.cpp.o] Error 1
CMakeFiles/Makefile2:117: recipe for target 'PyOpenPoseLib/CMakeFiles/PyOpenPose.dir/all' failed
make[1]: *** [PyOpenPoseLib/CMakeFiles/PyOpenPose.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2

```

**Comment by padeler** [comment]
Did you delete the cmake cache after the changes in the CMakeLists and rerun cmake?

**Comment by olimoser** [comment]
yes. I deleted the build folder including the cmake cache.

**Comment by padeler** [comment]
Did you find a solution for the issue? I was not able to reproduce it.

**Comment by christian-lanius** [comment]
I fixed the issue running with boost 1.66 as well. A dockerfile containing my build process can be found [here](https://gist.github.com/christian-lanius/fd6846d44eeee63719858c2291c559ba). As far as I can tell PyOpenPose works with the latest master commit from [openpose ( f6ade54)](https://github.com/CMU-Perceptual-Computing-Lab/openpose/commit/7325aa32dce312539e7414c1ba599631c3ad221b)
I had the additional problem that the numpy headers were not added correctly by cmake. Because I wasn't sure if this problem is due to my setup, I haven't included the fix in the pull request. The fix can be seen in the commit [ 16695cc](https://github.com/christian-lanius/PyOpenPose/commit/1b66c429703c3c265e2e234da9fd6e8d925529a9)
